contrib/internal/httptrace/config.go

```diff
@@ -39,10 +40,16 @@ type config struct {
 func newConfig() config {
 	c := config{
 		clientIPHeader:    os.Getenv(envClientIPHeader),
-		clientIP:          !internal.BoolEnv(envClientIPHeaderDisabled, false),
+		clientIP:          !internal.BoolEnv(envClientIPHeaderDisabled, true),
 		queryString:       !internal.BoolEnv(envQueryStringDisabled, false),
 		queryStringRegexp: defaultQueryStringRegexp,
 	}
+	// IP collection is disabled by default without ASM. If ASM is enabled, only disable collection if specifically
+	// asked through the env.
+	if appsec.Enabled() {
+		_, set := os.LookupEnv(envClientIPHeaderDisabled)
+		c.clientIP = c.clientIP || !set
+	}
```

Comment:
This won't fit well with RC, but I'm okay assuming no one will remotely enable Go's ASM for now (CGO, build tag, etc.). But we need at least a TODO here please to make this dynamic at some point, this means otherwise that people remotely enabling ASM won't get their client IP addresses being reported.

It's actually enough of a big deal to check if there's an easy solution right now.